src/components/GlobalDialog/index.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { inject, observer } from 'mobx-react';
+import { withStyles, Typography, Button, Dialog, DialogTitle, DialogContent, DialogActions } from '@material-ui/core';
+import { Error } from '@material-ui/icons';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+import styles from './styles';
+
+@injectIntl
+@withStyles(styles, { withTheme: true })
+@inject('store')
+@observer
+export default class GlobalDialog extends Component {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { classes, intl } = this.props;
+    const { globalDialog } = this.props.store.components;
+    const { visible, title, heading, body } = globalDialog;
+
+    return visible && (
+      <Dialog open={visible} onClose={globalDialog.reset}>
+        <DialogTitle>
+          <div className={classes.titleContainer}>
+            <Error className={classes.titleIcon} />
+            {title.id ? intl.formatMessage({ id: title.id, defaultMessage: title.defaultMessage }) : title}
```

Comment:
feel like if we go this way, without `defineMessage`, the extract script is kind of useless, we need to manually add the string to all language files, if we support more lanugaes, it will be likely to miss translation.

Reply:
sorry forgot about having the need to defineMessages. I will add this in.